packages/devtools/lib/src/ui/ui_utils.dart

```diff
@@ -224,6 +224,43 @@ class ServiceExtensionSelector {
   }
 }
 
+class SamplePeriodSelector {
+  SamplePeriodSelector() {
+    selector = PSelect()
+      ..small()
+      ..clazz('on-page-selector')
+      ..change(_handleSelect)
+      ..tooltip = 'The frequency at which the VM will collect CPU samples. For '
+          'a finer-grained profile, choose a smaller sample period. Please '
+          'read our documentation to understand the tradeoffs associated with '
+          'this setting.'
+      ..option('Sample period: 1000 μs', value: '1000')
```

Comment:
Let us tell people the tradeoffs here rather than telling them to look somewhere else or at least provide a link to exactly the documentation that explains the tradeoff.
What is missing is telling people why they should not always use a smaller sample period.
Maybe we should just having a checkbox for "Fine grained samples" or have a slider that lets you go from fine to coarse as these values seem pretty arbitrary.

Reply:
I like the option of having named entries instead of telling users what the sample rate was precisely.